tests/test_datacard.py

```diff
@@ -0,0 +1,47 @@
+import audbcards
+import pytest
+import difflib
+import numpy as np
+
+from audbcards.core.dataset import create_datasets_page
+
+
+def test_datacard_lines_similar(db, default_template):
+    """Create datacard using jinja2 via Dataset and Datacard.
+
+    The assertions for exact identity are currently too strict.
+    Therefore this uses text similarities as obtained from
+    the difflib builtins. These are based on
+
+    - average (or rather median and mean) similarities per line
+    - percentage of lines differing between original and rendered
+
+    """
+
+    dataset = audbcards.Dataset(pytest.NAME, pytest.VERSION)
+    dc = audbcards.Datacard(dataset)
+    content = dc._render_template()
+
+    ratios = []
+    for line1, line2 in zip(default_template.split("\n"), content.split("\n")):
+        sm = difflib.SequenceMatcher(lambda x: x == " ", line1, line2)
+        ratios.append(sm.ratio())
+
+    # there are lines that will differ (author, date of dataset)
+    # almost all lines are identical, so mean and median should be high.
+    assert np.median(np.array(ratios)) == 1
+    assert np.mean(np.array(ratios)) > 0.95
+
+    # use differ to check how many lines are identical
+    differ = difflib.Differ()
+    obtained, expected = content.splitlines(), default_template.splitlines()
+    result = list(differ.compare(obtained, expected))
+    # more than 90% of the lines should be identical
+    ratio_diff = len([x for x in result if x.startswith("-")])
+    assert 1. - ratio_diff / len(result) > 0.85
```

Comment:
This can be replaced by

```suggestion
content = content.rstrip()

for pattern in [
        re.compile('^published.*$', flags=(re.MULTILINE)),
        re.compile('^repository.*$', flags=(re.MULTILINE)),
]:
    content = re.sub(pattern, '', content)
    default_template = re.sub(pattern, '', default_template)

assert content == default_template
```

You will also have to add `import re` at the top of the file
and you need to update `tests/test_data/rendered_templates/default.rst` and replace `f0.wav` by `f1.wav`.

Afterwards, the test still fails, but because we actually have errors in `content`.
Here is the full diff as returned by `pytest -vv`

```
E           .. _db:                   
E                                                               
E           db                                                    
E           --                                                         
E                                                                                                                                                                                              
E           Created by H Wierstorf, C Geng, B E Abrougui                                                                                                                                       
E                                                                                                                                                                                              
E           ============= ======================                                                                                                                                               
E           version       `1.0.0 <https://github.com/audeering/db/blob/main/CHANGELOG.md>`__                                                                                                   
E           license       `CC0-1.0 <https://creativecommons.org/publicdomain/zero/1.0/>`__
E           source        https://github.com/audeering/audbcards               
E           usage         unrestricted                
E         - languages     eng                       
E         + languages     eng                                                                                                                                                                  
E         ?                  +      
E           format        wav                                   
E           channel       1                                                    
E           sampling rate 8000                             
E           bit depth     16                                                                                                                                                                   
E           duration      0 days 00:00:04                                                                                 
E           files         2                                                                                                                                                                    
E                                                                                       
E                                                                                                                                                                                              
E           ============= ======================
E           
E           Description
E           ^^^^^^^^^^^
E           
E           Example database.
E           
E           Example
E           ^^^^^^^
E           
E           :file:`data/f0.wav`
E           
E           .. image:: ../db.png
E           
E           .. raw:: html
E           
E               <p><audio controls src="db/data/f0.wav"></audio></p>
E           
E           Tables
E           ^^^^^^
E           
E           .. csv-table::
E              :header: ID,Type,Columns
E              :widths: 20, 10, 70
E           
E         -     "files", "filewise", "speaker"
E         +     "files", "filewise",  "speaker"
E         ?                          +
E         -     "segments", "segmented", "emotion"
E         +     "segments", "segmented",  "emotion"
E         ?                              +
E         -     "speaker", "misc", "age, gender"
E         +     "speaker", "misc",  "age, gender"
E         ?                        +
E           
E           Schemes
E           ^^^^^^^
E           
E           .. csv-table::
E              :header: ID,Dtype,Min,Labels,Mappings
E           
E               "age", "int", "", ""
E               "emotion", "str", "", "angry, happy, neutral"
E               "gender", "str", "", "female, male"
E               "speaker", "int", "", "0, 1", "age, gender"
```

So the problems are:

* In `content` we get an extra space at the end of languages: `\nlanguages     eng \n`
* In `content` we get an extra space in the CSV file listing the tables between the `Type` and `Columns` entries, e.g. `"files", "filewise",  "speaker"`

So, I think we should add this test and try to solve the issues, that let it fail.

Reply:
To make it easier to see I created https://github.com/audeering/audbcards/pull/23 that includes all my proposed/mentioned changes, so you could also merge from their instead of applying single suggestions here.

Reply:
This is merged into this branch now.